spec/models/season/phase_switcher_spec.rb

```diff
@@ -4,32 +4,61 @@
 context 'when switching phases' do
   subject { Season.current }
 
-  describe '#fake_application_phase' do
-    it 'timeshifts the application phase to today' do
-      Timecop.travel(Season.current.applications_close_at - 1.month) do
-        Season::PhaseSwitcher.fake_application_phase
-        expect(subject).to be_application_period
+  describe 'destined' do
+
+    it 'sends the params to a phase setting method' do
+      phase = 'fake_application_phase'
+      Season::PhaseSwitcher.destined(phase)
+      expect(subject).to_not be_nil
+    end
+
+    context 'when directed to the #fake_application_phase' do
+      it 'sets date range so that the season is open for application submissions' do
+        phase = 'fake_application_phase'
+        Timecop.travel(Season.current.applications_close_at - 1.month) do
+          Season::PhaseSwitcher.destined(phase)
+          expect(subject).to_not be_nil
+          expect(subject).to be_application_period
+        end
       end
     end
-  end
 
-  describe '#fake_coding_phase' do
-    it 'timeshifts the coding phase to today' do
-      Timecop.travel(Season.current.ends_at - 1.week) do
-        Season::PhaseSwitcher.fake_coding_phase
-        expect(subject).to be_started
+    context 'when directed to the #fake_coding_phase' do
+      it 'sets dates so that the summer of code is currently happening' do
+        phase = 'fake_coding_phase'
+        Timecop.travel(Season.current.ends_at - 1.week) do
```

Comment:
I am very bad at writing tests and writing Rspec, sorry. IIRC I _intended_ to test the interface only, and now I understand I did not. 
So, I tried your suggestions and I don't understand what to do. The `expect and receive` things looks to me as it is exactly what we need, I didn't know about that. But I got the strangest errors, never got past the first part where the Season is expected to receive :current. So, maybe you can give me hint in the right direction?
Or I can show a few things I tried? Whatever is easier for you. 
So, I moved to part 2, because it makes totally sense that the season should not be the subject. Here too: all kind of errors. I did a sanity check: 
```
it "is available as described_class" do
expect(described_class).to eq(PhaseSwitcher)
end
```
and that failed. 🤔  
Tips?